tests/test_mongodb_table_reset_inc.py

```diff
@@ -0,0 +1,298 @@
+                self.assertSetEqual(replication_key, expected_replication_keys)
+
+        #  -----------------------------------
+        #  ----------- Initial Sync ----------
+        #  -----------------------------------
+
+        # Select simple_coll_1 and simple_coll_2 streams and add replication method metadata
+        for stream_catalog in found_catalogs:
+            annotated_schema = menagerie.get_annotated_schema(conn_id, stream_catalog['stream_id'])
+            rep_key = 'date_field'
+            for key in self.key_names():
+                if key in stream_catalog['stream_name']:
+                    rep_key = key
+            additional_md = [{ "breadcrumb" : [], "metadata" : {'replication-method' : 'INCREMENTAL',
+                                                                'replication-key': rep_key}}]
+            selected_metadata = connections.select_catalog_and_fields_via_metadata(conn_id,
+                                                                                   stream_catalog,
+                                                                                   annotated_schema,
+                                                                                   additional_md)
+
+        # Run sync
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        # verify the persisted schema was correct
+        messages_by_stream = runner.get_records_from_target_output()
+
+        # gather expectations
+        expected_schema = {'type': 'object'}
+
+        for tap_stream_id in self.expected_sync_streams():
+            with self.subTest(stream=tap_stream_id):
+
+                # gather results
+                persisted_schema = messages_by_stream[tap_stream_id]['schema']
+
+                # assert the schema is an object
+                self.assertDictEqual(expected_schema, persisted_schema)
+
+        # verify that each of the streams that we synced are the ones that we expect to see
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        # verify that the entire collection was synced by comparing row counts against the source
+        for tap_stream_id in self.expected_sync_streams():
+            with self.subTest(stream=tap_stream_id):
+
+                expected_row_count = self.expected_row_counts()[tap_stream_id]
+                row_count = record_count_by_stream[tap_stream_id]
+
+                self.assertEqual(expected_row_count, row_count)
+
+        # verify state is saved in the proper format for all streams
```

Comment:
Instead of doing these assertions after sync 1, can you do them for sync 2? I think the standard incremental test should be covering the assertions with this setup. But the new scenario you introduced with the table reset should be covered.

Reply:
Will do